File: algorithms/python/Challenge_Sort_Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        redIdx, blueIdx = 0, len(nums) - 1
        i = 0
        
        while i <= blueIdx:
                
            if nums[i] == 0:
                nums[redIdx], nums[i] = nums[i], nums[redIdx]
                i += 1
                redIdx += 1
                
            elif nums[i] == 2:
                nums[blueIdx], nums[i] = nums[i], nums[blueIdx]
                blueIdx -= 1
                
            else:
                i += 1

File: algorithms/java/Challenge_Sort_Colors.java
class Solution {
    public void sortColors(int[] nums) {
        int redIdx = 0;
        int blueIdx = nums.length - 1;
        int currIdx = 0;
        
        while(currIdx <= blueIdx) {
            if(nums[currIdx] == 0) {
                nums[currIdx] = nums[redIdx];
                nums[redIdx] = 0;
                ++currIdx; ++redIdx;
            } else if(nums[currIdx] == 2) {
                nums[currIdx] = nums[blueIdx];
                nums[blueIdx] = 2;
                --blueIdx;
            } else {
                ++currIdx;
            }
        }
    }
}
